src/master-playlist-controller.js

```diff
@@ -794,23 +794,20 @@ export class MasterPlaylistController extends videojs.EventTarget {
       return this.mediaSource.endOfStream('network');
     }
 
-    // Blacklist this playlist
-    currentPlaylist.excludeUntil = Date.now() + BLACKLIST_DURATION;
+    if (!this.masterPlaylistLoader_.isFinalRendition_()) {
+      // Blacklist this playlist
+      currentPlaylist.excludeUntil = Date.now() + BLACKLIST_DURATION;
 
-    // Select a new playlist
-    nextPlaylist = this.selectPlaylist();
+      // Select a new playlist
+      nextPlaylist = this.selectPlaylist();
 
-    if (nextPlaylist) {
       videojs.log.warn('Problem encountered with the current ' +
                        'HLS playlist. Switching to another playlist.');
 
       return this.masterPlaylistLoader_.media(nextPlaylist);
     }
-    videojs.log.warn('Problem encountered with the current ' +
-                     'HLS playlist. No suitable alternatives found.');
-    // We have no more playlists we can select so we must fail
-    this.error = error;
-    return this.mediaSource.endOfStream('network');
+    // Never blacklisting this playlist because it's final rendition
+    return this.masterPlaylistLoader_.media(currentPlaylist);
```

Comment:
No I don't think that will work. When a playlist request errors, `PlaylistLoader` triggers an error and returns without setting up another request. At this point we want to kick `PlaylistLoader` back into gear to start requesting the final rendition again. Calling `media()` will normally do this as long as the media passed in is not the same as the current media. Calling `load()` will also kickstart the loader, but depending on the loader's `state` after an error occurred, a `load` may never actually get the loader started. This will probably require some changes to the way `PlaylistLoader` handles errors so after an error, it gets reset to a state that it can start loading again